snappi_ixnetwork/resourcegroup.py

```diff
@@ -36,25 +27,36 @@ def set_group(self):
             response = self._api._request('POST', url, payload)
         except Exception:
             raise Exception("Not able to fetch chassis details. Unable to execute L1 setting")
-        
-        self._process_property()
-        try:
-            chassis_id = 1
-            result = response['result'][0]
-            self._process_groups(result['cards'], chassis_id)
-        except Exception:
-            raise Exception("Problem to parse chassis details")
+        chassis_id = 1
+        for result in response['result']:
+            chassis_dns = result["dns"]
```

Comment:
does key "dns" always present in the result? if key "dns" is not available it will throw key error.
avoid it by result.get("dns")

Reply:
I found "dns" present most of our chassis and appliance. And as this is our internal IxNetwok call probably that should applicable for other type. Anyway I have global exception block which will handle these kind of excption.